components/camel-kafka/src/main/java/org/apache/camel/processor/idempotent/kafka/KafkaIdempotentRepository.java

```diff
@@ -0,0 +1,342 @@
+ * consumer group.
+ *
+ * On startup, the instance subscribes to the topic and rewinds the offset to the beginning, rebuilding the cache to the
+ * latest state.
+ *
+ * To use, this repository must be placed in the Camel registry, either manually or by registration as a bean in
+ * Spring/Blueprint, as it is CamelContext aware.
+ *
+ * @author jkorab
+ */
+@ManagedResource(description = "Kafka IdempotentRepository")
+public class KafkaIdempotentRepository extends ServiceSupport implements IdempotentRepository<String>, CamelContextAware {
+
+    private static final int DEFAULT_MAXIMUM_CACHE_SIZE = 1000;
+
+    private final Logger log = LoggerFactory.getLogger(this.getClass());
+    private final Map<String, Object> cache;
+
+    private final AtomicLong duplicateCount = new AtomicLong(0);
+    private final String topic;
+    private final Properties producerConfig;
+    private final Properties consumerConfig;
+
+    private Consumer<String, String> consumer;
+    private Producer<String, String> producer;
+    private TopicPoller topicPoller;
+
+    private CamelContext camelContext;
+    private ExecutorService executorService;
+    private CountDownLatch cacheReadyLatch;
+
+    enum CacheAction {
+        add,
+        remove,
+        clear
+    }
+
+    public KafkaIdempotentRepository(String topic, String bootstrapServers) {
+        this(topic, bootstrapServers, DEFAULT_MAXIMUM_CACHE_SIZE);
+    }
+
+    public KafkaIdempotentRepository(String topic, String bootstrapServers, int maxCacheSize) {
+        StringHelper.notEmpty(topic, "topic");
+        StringHelper.notEmpty(bootstrapServers, "bootstrapServers");
+        Properties consumerConfig = new Properties();
+        consumerConfig.put(ConsumerConfig.BOOTSTRAP_SERVERS_CONFIG, bootstrapServers);
+
+        Properties producerConfig = new Properties();
+        producerConfig.put(ProducerConfig.BOOTSTRAP_SERVERS_CONFIG, bootstrapServers);
+
+        this.topic = topic;
+        this.consumerConfig = consumerConfig;
+        this.producerConfig = producerConfig;
+        this.cache = Collections.synchronizedMap(new LRUCache<>(maxCacheSize));
+    }
+
+    public KafkaIdempotentRepository(String topic, Properties consumerConfig, Properties producerConfig) {
+        this(topic, consumerConfig, producerConfig, DEFAULT_MAXIMUM_CACHE_SIZE);
+    }
+
+    public KafkaIdempotentRepository(String topic, Properties consumerConfig, Properties producerConfig, int maxCacheSize) {
+        StringHelper.notEmpty(topic, "topic");
+        this.topic = topic;
+        ObjectHelper.notNull(consumerConfig, "consumerConfig");
+        this.consumerConfig = consumerConfig;
+        ObjectHelper.notNull(producerConfig, "producerConfig");
+        this.producerConfig = producerConfig;
+        this.cache = Collections.synchronizedMap(new LRUCache<>(maxCacheSize));
+    }
+
+    @Override
+    protected void doStart() throws Exception {
+        log.info("Context: {}", camelContext);
```

Comment:
A better logging message would be nice. Something about starting XXX with YY setting (the important values)